internal/consensus/state.go

```diff
@@ -2070,20 +2059,20 @@ func (cs *State) addVote(vote *types.Vote, peerID types.NodeID) (added bool, err
 		prevotes := cs.Votes.Prevotes(vote.Round)
 		cs.Logger.Debug("added vote to prevote", "vote", vote, "prevotes", prevotes.StringShort())
 
-		// If +2/3 prevotes for a block or nil for *any* round:
-		if blockID, ok := prevotes.TwoThirdsMajority(); ok {
-			// There was a polka!
-			// If we're locked but this is a recent polka, unlock.
-			// If it matches our ProposalBlock, update the ValidBlock
+		// Check to see if >2/3 of the voting power on the network voted for any non-nil block.
```

Comment:
Side comment: we should probably add invariant check that `if len(blockID.Hash) != 0 ` at the beginning of this function, for example at line 1535. It is currently done at the caller, but we should probably have it explicitly also inside this function so we don't rely only on caller to do it properly.

Reply:
I think I'm a little confused by which function you're referring to. This code here is within `addVote`, which doesn't seem to require that there be a non-nil polka yet.

Reply:
It is `enterCommit` function, line 1535.

Reply:
I see, I'm going to create a github issue to track this. I'd prefer to keep this commit somewhat focused so that it's easier to roll back in the event that it causes any issue.